--- a/plugins/quickinspector/quickinspector.cpp
+++ b/plugins/quickinspector/quickinspector.cpp
@@ -662,7 +662,7 @@ void QuickInspector::checkOverlaySettings()
 class SGSoftwareRendererPrivacyViolater : public QSGAbstractSoftwareRenderer
 {
 public:
-#if QT_VERSION >= QT_VERSION_CHECK(5, 12, 0)
+#if QT_VERSION >= QT_VERSION_CHECK(5, 12, 0) && QT_VERSION != QT_VERSION_CHECK(5, 14, 0)
     using QSGAbstractSoftwareRenderer::renderableNodes;
 #endif
     using QSGAbstractSoftwareRenderer::renderNodes;
@@ -692,7 +692,7 @@ void QuickInspector::analyzePainting()
         renderer->markDirty();
         renderer->buildRenderList();
         renderer->optimizeRenderList();
-#if QT_VERSION < QT_VERSION_CHECK(5, 12, 0)
+#if QT_VERSION < QT_VERSION_CHECK(5, 12, 0) || QT_VERSION == QT_VERSION_CHECK(5, 14, 0)
         renderer->renderNodes(&painter);
 #else
         auto iterator = renderer->renderableNodes().begin();
